pkg/scheduler/core/generic_scheduler.go

```diff
@@ -918,7 +918,10 @@ func filterPodsWithPDBViolation(pods []*v1.Pod, pdbs []*policy.PodDisruptionBudg
 					pdbForPodIsViolated = true
 					break
```

Comment:
I think the previous logic is that if the pod violates one pdb, then It will be added to the `violatingpods`. Then we don't need to check more pdbs here. @ahg-g

Reply:
We still need to take into account its impact on other PDBs though. Consider the case where the pod matches another pdb that hasn't been violated yet, and so we need to decrement podsAllowed for that one.

Reply:
got it. Thanks. I remove `break` here.